sdks/python/apache_beam/transforms/core.py

```diff
@@ -220,15 +222,29 @@ def __init__(self, generator, *args, **kwargs):
     self._gen_args = args
     self._gen_kwargs = kwargs
 
+  def element_generator(self):
+    while self._current_value is not None:
+      yield self._current_value
+
+    raise StopIteration()
+
   def start_bundle(self, context):
-    self._generator = self._generator_def(*self._gen_args, **self._gen_kwargs)
-    self._generator.send(None)
+    self._generator = self._generator_def(
+        self.element_generator(), *self._gen_args, **self._gen_kwargs)
 
   def process(self, context, *args, **kwargs):
-    return self._generator.send((context, args, kwargs))
+    self._current_value = (context, args, kwargs)
```

Comment:
The way I read this, the generator must have a 1:1 relationship between consuming elements of the element iterable and yielding them. In particular, it's silent data loss/duplication to not do so.